src/main/java/com/philips/swcoe/cerberus/cerebellum/codemetrics/java/CodeMetricsVerticalWriterService.java

```diff
@@ -14,59 +14,56 @@
-            getStreamMetricsMethods(javaCodeMetricsClassResult.getClass().getMethods()).forEach(methodInClassResult -> {
-                Try.of(() -> (CodeMetricsResult) methodInClassResult.invoke(javaCodeMetricsClassResult))
-                                    .andThen(codeMetricsResultForClass -> Try.run(() -> writeClassMetrics(csvPrinter, javaCodeMetricsClassResult, codeMetricsResultForClass ))
-                                    .onFailure(exception -> log.trace(exception.getStackTrace())));
-            });
-            List<CodeMetricsMethodResult> codeMetricsMethodResults = javaCodeMetricsClassResult.getMethodMetrics();
+        codeMetricsClassResults.stream().forEach(codeMetricsClassResult -> {
+            processMetrics(codeMetricsClassResult, classConfig);
+            List<CodeMetricsMethodResult> codeMetricsMethodResults = codeMetricsClassResult.getMethodMetrics();
             codeMetricsMethodResults.stream().forEach(codeMetricsMethodResult -> {
-                getStreamMetricsMethods(codeMetricsMethodResult.getClass().getMethods()).forEach(methodInMethodResult -> {
-                    Try.of(() -> (CodeMetricsResult) methodInMethodResult.invoke(codeMetricsMethodResult))
-                                    .andThen(codeMetricsResultForMethod -> Try.run(() -> writeMethodMetrics(csvPrinter, codeMetricsMethodResult, codeMetricsResultForMethod, javaCodeMetricsClassResult))
-                                    .onFailure(exception -> log.trace(exception.getStackTrace())));
-                });
+                processMetrics(codeMetricsMethodResult, methodConfig);
             });
         });
         return this.getReport();
     }
 
-    private void writeClassMetrics(CSVPrinter csvPrinter,
-                                   CodeMetricsClassResult codeMetricsClassResult,
-                                   CodeMetricsResult codeMetricsResult) throws IOException {
-        List<String> classMetricsToDisplay = getMetricsToDisplayFromConfig(classConfig);
-        if(doesItMatterToDisplay(codeMetricsResult, classMetricsToDisplay)) {
-            csvPrinter.printRecord(
-                    codeMetricsClassResult.getFile(),
-                    codeMetricsClassResult.getClassName(),
-                    codeMetricsClassResult.getType(),
-                    codeMetricsResult.getMetricName(),
-                    codeMetricsResult.getNewValue(),
-                    codeMetricsResult.getOldValue()
-            );
+    private void processMetrics(CodeMetricsResult codeMetricsResult, List<String> config) {
+        getStreamMetricsMethods(codeMetricsResult.getClass().getMethods()).forEach(methodInResult -> {
+            Try.of(() -> (CodeMetricsDiffResult) methodInResult.invoke(codeMetricsResult))
+                                .andThen(codeMetricsResultForConstruct -> Try.run(() -> writeMetrics(codeMetricsResultForConstruct, config))
+                                .onFailure(exception -> log.trace(exception.getStackTrace())));
```

Comment:
Thanks for identifying this, I am tracing because it helps us to support people if they share the trace of program flow. Since I am handling exception in monadic way, It becomes hard for us to support if some one shares just an exception .